File: java/src/main/java/gov/doc/ntia/sigmf/ext/global/waveform/CodingRate.java
package gov.doc.ntia.sigmf.ext.global.waveform;

import com.fasterxml.jackson.annotation.JsonAnyGetter;
import com.fasterxml.jackson.annotation.JsonAnySetter;
import com.fasterxml.jackson.annotation.JsonInclude;
import java.io.Serializable;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class CodingRate extends ArrayList<Long> implements Serializable {

  private static final long serialVersionUID = 1L;

  protected Map<String, Object> otherFields = new HashMap<>();

  public CodingRate() {

    super(2);
  }

  public void setK(long k) {
    add(0, k);
  }

  public void setN(long n) {
    add(1, n);
  }

  @JsonAnyGetter
  public Map<String, Object> getOtherFields() {
    return otherFields;
  }

  @JsonAnySetter
  public void add(String key, Object value) {
    otherFields.put(key, value);
  }
}

File: Csharp/sigmf/sigmf/ext/global/waveform/CodingRate.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Text;

namespace sigmf
{
    [Serializable()]
    public class CodingRate : ArrayList

        // fix the list
    {
        private static readonly long serialVersionUID = 1L;

        protected Dictionary<string, Object> otherfields = new Dictionary<string, Object>();

        public CodingRate() {
            //this.base(2);
        }
        public void setK(long k) {
            this.Insert(0,k);
        }
        public void setN(long n) {
            this.Insert(1,n);
        }
        public Dictionary<string, object> getOtherFields()
        {
            return otherfields;
        }
        public void add(string key, object value)
        {
            otherfields.Add(key, value);
        }
    }
}
